src/modules/Financial/models/wallet.model.ts

```diff
@@ -0,0 +1,100 @@
+import prisma from '@/config/client';
+import { handlePrismaError } from '@/errors';
+import type { wallets } from '@/generated/prisma';
+import type { Wallet, CreateWalletData, UpdateWalletData } from '../types';
+
+// Helper to DRY
+const transformWallet = (wallet: wallets): Wallet => ({
+  ...wallet,
+  owner_id: wallet.owner_id!,
+  wallet_type: wallet.wallet_type as 'individual' | 'organization',
+  balance: Number(wallet.balance || 0),
+  status: wallet.status as 'active' | 'suspended',
+});
+
+// Wallet operations
+const createWallet = async (
+  ownerId: number,
+  data: CreateWalletData
+): Promise<Wallet> => {
+  try {
+    const wallet = await prisma.wallets.create({
+      data: {
+        owner_id: ownerId,
+        wallet_type: data.wallet_type,
+        organization_type: data.organization_type || null,
+        balance: 0,
+        status: 'active',
+      },
+    });
+    return transformWallet(wallet);
+  } catch (error) {
+    handlePrismaError(error);
+  }
+};
+
+const findWalletsByUserId = async (userId: number): Promise<Wallet[]> => {
+  try {
+    const wallets = await prisma.wallets.findMany({
+      where: { owner_id: userId },
+    });
+    return wallets.map(transformWallet);
+  } catch (error) {
+    handlePrismaError(error);
+  }
+};
+
+const findWalletById = async (id: number): Promise<Wallet | null> => {
+  try {
+    const wallet = await prisma.wallets.findUnique({
+      where: { id },
+    });
+    return wallet ? transformWallet(wallet) : null;
```

Comment:
the business logic is alrdy in [here](https://github.com/CSC290-2025/backend/pull/12/files#diff-ffcfe541fae8fa20e621cd28ce8599e5f2097f0bd4937101c64e3dc76cfeaffbR19) right?  I don't think you need a ternary operator. Let me know if my reasoning is incorrect.

Reply:
chai